File: easy/4_SumOfPrimes/ruby/lib/sum_of_primes.rb
class SumOfPrimes
  def initialize(num_of_primes)
    @num_of_primes = num_of_primes
    puts first_n_primes.inject(&:+)
  end

  def first_n_primes
    return_array = Array.new
    num = 2
    while (return_array.length < @num_of_primes)
      return_array << num if is_prime?(num)
      num +=1
    end
    return_array
  end

  def is_prime?(num)
    2.upto(Math.sqrt(num)) do |val|
      return false if num % val == 0
    end
    true
  end
end
#SumOfPrimes.new(1000)

File: easy/004_SumOfPrimes/go/sum_of_primes.go
package main

import "fmt"
import "math"

func main() {
	fmt.Println(SumOfPrimes(1000))
}

func SumOfPrimes(count int) int {
	primes := FirstNPrimes(count)
	sum := 0

	for _, prime := range primes {
		sum += prime
	}

	return sum
}

func FirstNPrimes(n int) []int {
	pp := make([]int, n)
	c := 1
	i := 0

	for p := 2; c <= n; p++ {
		if IsPrime(p) {
			pp[i] = p
			i++
			c++
		}
	}

	return pp
}

func IsPrime(p int) bool {
	prime := true

	for i := 2; float64(i) <= math.Sqrt(float64(p)); i++ {
		if math.Mod(float64(p), float64(i)) == 0 {
			prime = false
			break
		}
	}

	return prime
}
